sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -579,7 +579,7 @@ class SQLContext private[sql](
     val className = beanClass.getName
     val beanInfo = Introspector.getBeanInfo(beanClass)
     val rows = SQLContext.beansToRows(data.asScala.iterator, beanInfo, attrSeq)
-    DataFrame(self, LocalRelation(attrSeq, rows.toSeq))
+    DataFrame(self, LocalRelation(attrSeq, rows.toArray))
```

Comment:
toArray could be kind of expensive for large local iterators - maybe we should only do this if necessary?